### ISSUE TITLE
How to combine ty's suppression with Ruff's?

### ISSUE THREAD

**Issue by hynek** [astral-sh/ty#1467]
### Question

I have [code](https://github.com/hynek/svcs/blob/1d6efae1a517eeabc3d283212e6e4bb058a10d85/src/svcs/_core.py#L484
) in _svcs_ that needs to be a bit clever about it treats callables in order to support both sync and async.

I have to suppress PLW2901 (redefinition of loop variable), but with some recent-ish ty release I now also have to suppress `call-non-callable`.

The docs [say](https://docs.astral.sh/ty/suppression/#ty-suppression-comments):

> To suppress a rule violation inline add a `# ty: ignore[<rule>]` comment **at the end of the line**

So I hoped I could just write:

```python
                if iscoroutinefunction(oc):
                    oc = oc()  # noqa: PLW2901 ty: ignore[call-non-callable]

```

but just like Ruff, ty ignores the second part. I can only suppress one: the first one.

What am I missing? I'd expect the case of where I have to silence both the linter and the type checker is quite common, but I haven't found anything on the bug tracker.

I literally don't know what to do, so I've disabled ty for now again.

### Version

ty 0.0.1-alpha.25 (3abd4c968 2025-10-29)

**Comment by MichaReiser** [comment]
Thanks for the nice write up. 

Either of the following two should work:

```py
                if iscoroutinefunction(oc):
                    oc = oc()  # noqa: PLW2901 # ty: ignore[call-non-callable]
```

```py
                if iscoroutinefunction(oc):
                    oc = oc()  # ty: ignore[call-non-callable] # noqa: PLW2901 
```

The key is to start each `noqa` or `ty: ignore` comment with its own `#`. 

[Here](https://github.com/astral-sh/ruff/blob/main/crates/ty_python_semantic/resources/mdtest/suppressions/type_ignore.md#nested-comments) a few examples from our test suite.

Given that this has come up before on the ruff side, I think it does make sense to add an example like this to the documentation

**Comment by hynek** [comment]
Ah wow yes, that works and with pyrefly too – and nobody has documented it. :D  Thanks!
